components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -36,92 +36,95 @@ const SidebarChannelMenu = (props: Props) => {
+                onKeyDown={handleMarkAsRead}
+                onMouseDown={handleMarkAsRead}
             />
 
         );
     } else {
-        function handleMarkAsUnread(event: MouseEvent<HTMLLIElement>) {
-            event.preventDefault();
-
-            props.markMostRecentPostInChannelAsUnread(props.channel.id);
-            trackEvent('ui', 'ui_sidebar_channel_menu_markAsUnread');
+        function handleMarkAsUnread(event: MouseEvent<HTMLLIElement> | KeyboardEvent<HTMLLIElement>) {
+            if (Menu.isPressed(event)) {
+                props.markMostRecentPostInChannelAsUnread(props.channel.id);
+                trackEvent('ui', 'ui_sidebar_channel_menu_markAsUnread');
+            }
         }
 
         markAsReadUnreadMenuItem = (
             <Menu.Item
                 id={`markAsUnread-${props.channel.id}`}
-                onClick={handleMarkAsUnread}
                 leadingElement={<MarkAsUnreadIcon size={18}/>}
                 labels={(
                     <FormattedMessage
                         id='sidebar_left.sidebar_channel_menu.markAsUnread'
                         defaultMessage='Mark as Unread'
                     />
                 )}
+                onKeyDown={handleMarkAsUnread}
+                onMouseDown={handleMarkAsUnread}
             />
         );
     }
 
     let favoriteUnfavoriteMenuItem: JSX.Element | null = null;
     if (props.isFavorite) {
-        function handleUnfavoriteChannel(event: MouseEvent<HTMLLIElement>) {
-            event.preventDefault();
-
-            props.unfavoriteChannel(props.channel.id);
-            trackEvent('ui', 'ui_sidebar_channel_menu_unfavorite');
+        function handleUnfavoriteChannel(event: MouseEvent<HTMLLIElement> | KeyboardEvent<HTMLLIElement>) {
+            if (Menu.isPressed(event)) {
+                props.unfavoriteChannel(props.channel.id);
+                trackEvent('ui', 'ui_sidebar_channel_menu_unfavorite');
+            }
         }
 
         favoriteUnfavoriteMenuItem = (
             <Menu.Item
                 id={`unfavorite-${props.channel.id}`}
-                onClick={handleUnfavoriteChannel}
                 leadingElement={<StarIcon size={18}/>}
                 labels={(
                     <FormattedMessage
                         id='sidebar_left.sidebar_channel_menu.unfavoriteChannel'
                         defaultMessage='Unfavorite'
                     />
                 )}
+                onKeyDown={handleUnfavoriteChannel}
+                onMouseDown={handleUnfavoriteChannel}
             />
         );
     } else {
-        function handleFavoriteChannel(event: MouseEvent<HTMLLIElement>) {
-            event.preventDefault();
-
-            props.favoriteChannel(props.channel.id);
-            trackEvent('ui', 'ui_sidebar_channel_menu_favorite');
+        function handleFavoriteChannel(event: MouseEvent<HTMLLIElement> | KeyboardEvent<HTMLLIElement>) {
+            if (Menu.isPressed(event)) {
```

Comment:
Should this just be something we check in the menu item component before initiating the provided action? I don't think it's sustainable to do this for every menu item?

Reply:
Yeah, I agree with this as well. Ideally, this logic should be encapsulated inside the `MenuItem` rather than making whoever's using the menu do it

Reply:
encapsulated inside of menu item itself